### DISCUSSION TITLE
Custom Validation of flags

### DISCUSSION THREAD

**Discussion by shrimalmadhur** [urfave/cli#1954] [Q&A]
I am doing a lot of custom validation of the flags I use in my CLI App. I end up adding those validation in `Action` which kinds of bloats up the action. I am wondering what's the best way to do it. Should this be a feature on the flag itself that someone can add a custom validation code and then it runs while reading the flag value?

**Comment by dearchap** [comment]
@shrimalmadhur which version are you using ? v2 doesnt have any explicit validation support while v3 has support for specifying a validation function for each flag

**Reply by shrimalmadhur** [reply]
@dearchap ah I am using v2 since v3 is still alpha. I can wait for v3 to go out of alpha. When do you expect it to be formally released?
